configman/config_manager.py

```diff
@@ -147,19 +165,33 @@ def __init__(
         self.option_definitions = Namespace()
         self.definition_source_list = definition_source_list
 
+        command_line_value_source = command_line
         if values_source_list is None:
             # nothing set, assume defaults
             if use_admin_controls:
                 values_source_list = (
                     ConfigFileFutureProxy,
                     environment,
-                    command_line
+                    command_line_value_source
                 )
             else:
                 values_source_list = (
                     environment,
-                    command_line
+                    command_line_value_source
                 )
+        # determine which command_line facility to use for help
+        if self.use_auto_help:
+            for a_value_source in values_source_list:
+                if inspect.ismodule(a_value_source):
+                    handler = type_handler_dispatch[a_value_source][0].ValueSource
+                    try:
+                        if handler.command_line_value_source:
+                            handler._setup_auto_help(self)
+                            break
+                    except AttributeError:
+                        # not a commandline source, ok to ignore exception
```

Comment:
Which AttributeError is that? for `handler.command_line_value_source` or for `handler._setup_auto_help(self)`?